code/modules/mob/living/carbon/spirit/spirit.dm

```diff
@@ -182,19 +182,23 @@
 		return FALSE
 	var/success = FALSE
 	if(isliving(coffin))
-		success ||= pacify_corpse(coffin, user)
+		if(pacify_corpse(coffin, user))
+			success = TRUE
 	for(var/mob/living/corpse in coffin)
-		success ||= pacify_corpse(corpse, user)
+		if(pacify_corpse(corpse, user))
+			success = TRUE
 	for(var/obj/item/bodypart/head/head in coffin)
 		if(!head.brainmob)
 			continue
-		success ||= pacify_corpse(head.brainmob, user)
+		if(pacify_corpse(head.brainmob, user))
+			success = TRUE
 	//if this is a deep search, we will also search the contents of the coffin to pacify (EXCEPT MOBS, SINCE WE HANDLED THOSE)
 	if(deep)
 		for(var/atom/movable/stuffing in coffin)
 			if(isliving(stuffing) || istype(stuffing, /obj/item/bodypart/head))
 				continue
-			success ||= pacify_coffin(stuffing, user, deep, give_pq = FALSE)
+			if(pacify_coffin(stuffing, user, deep, give_pq = FALSE))
+				success = TRUE
 	// Success is actually the ckey of the last attacker so we can prevent PQ farming from fragging people
 	if(success && give_pq && user?.ckey && (user.ckey != success))
 		adjust_playerquality(give_pq, user.ckey)
```

Comment:
Doesn't you changing the return value of `pacify_coffin()` / `success` brick this check ( `TRUE != some ckey` is always TRUE)

(hi hi)